rust/lance-arrow/src/json.rs

```diff
@@ -0,0 +1,365 @@
+        decode_json(jsonb_bytes)
+            .map_err(|e| ArrowError::InvalidArgumentError(format!("Failed to decode JSON: {}", e)))
+    }
+
+    /// Get the value at index i as raw JSONB bytes
+    pub fn value_bytes(&self, i: usize) -> &[u8] {
+        self.inner.value(i)
+    }
+
+    /// Get JSONPath value from the JSON at index i
+    pub fn json_path(&self, i: usize, path: &str) -> Result<Option<String>, ArrowError> {
+        if self.inner.is_null(i) {
+            return Ok(None);
+        }
+
+        let jsonb_bytes = self.inner.value(i);
+        get_json_path(jsonb_bytes, path).map_err(|e| {
+            ArrowError::InvalidArgumentError(format!("Failed to extract JSONPath: {}", e))
+        })
+    }
+}
+
+impl Array for JsonArray {
+    fn as_any(&self) -> &dyn std::any::Any {
+        self
+    }
+
+    fn to_data(&self) -> ArrayData {
+        self.inner.to_data()
+    }
+
+    fn into_data(self) -> ArrayData {
+        self.inner.into_data()
+    }
+
+    fn data_type(&self) -> &DataType {
+        &DataType::LargeBinary
+    }
+
+    fn slice(&self, offset: usize, length: usize) -> ArrayRef {
+        Arc::new(Self {
+            inner: self.inner.slice(offset, length),
+        })
+    }
+
+    fn len(&self) -> usize {
+        self.inner.len()
+    }
+
+    fn is_empty(&self) -> bool {
+        self.inner.is_empty()
+    }
+
+    fn offset(&self) -> usize {
+        self.inner.offset()
+    }
+
+    fn nulls(&self) -> Option<&arrow_buffer::NullBuffer> {
+        self.inner.nulls()
+    }
+
+    fn get_buffer_memory_size(&self) -> usize {
+        self.inner.get_buffer_memory_size()
+    }
+
+    fn get_array_memory_size(&self) -> usize {
+        self.inner.get_array_memory_size()
+    }
+}
+
+// TryFrom implementations for string arrays
+impl TryFrom<StringArray> for JsonArray {
+    type Error = ArrowError;
+
+    fn try_from(array: StringArray) -> Result<Self, Self::Error> {
+        Self::try_from(&array)
+    }
+}
+
+impl TryFrom<&StringArray> for JsonArray {
+    type Error = ArrowError;
+
+    fn try_from(array: &StringArray) -> Result<Self, Self::Error> {
+        let mut builder = LargeBinaryBuilder::with_capacity(array.len(), array.value_data().len());
+
+        for i in 0..array.len() {
+            if array.is_null(i) {
+                builder.append_null();
+            } else {
+                let json_str = array.value(i);
+                let encoded = encode_json(json_str).map_err(|e| {
+                    ArrowError::InvalidArgumentError(format!("Failed to encode JSON: {}", e))
+                })?;
+                builder.append_value(&encoded);
+            }
+        }
+
+        Ok(Self {
+            inner: builder.finish(),
+        })
+    }
+}
+
+impl TryFrom<LargeStringArray> for JsonArray {
+    type Error = ArrowError;
+
+    fn try_from(array: LargeStringArray) -> Result<Self, Self::Error> {
+        Self::try_from(&array)
+    }
+}
+
+impl TryFrom<&LargeStringArray> for JsonArray {
```

Comment:
Hmm, supporting both string and large string is ok, but not sure it is vital at the moment if it becomes an inconvenience.